### PR TITLE
Adds custom encodings support to TransportRuntime.

### PR DESCRIPTION
**Pull request by vkryachko**

### CODE REVIEW

File: transport/transport-backend-cct/src/test/java/com/google/android/datatransport/cct/CctTransportBackendTest.java [hunk 1]
```diff
@@ -106,14 +109,16 @@ private BackendRequest getCCTBackendRequest(String transportName, CCTDestination
                         .setEventMillis(INITIAL_WALL_TIME)
                         .setUptimeMillis(INITIAL_UPTIME)
                         .setTransportName(transportName)
-                        .setPayload(PAYLOAD.toByteArray())
+                        .setEncodedPayload(
```
Comment (rlazo) [inline transport/transport-backend-cct/src/test/java/com/google/android/datatransport/cct/CctTransportBackendTest.java]: Should we test the case in which the payload isn't encoded in a supported format?

File: transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/EncodedPayload.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+import androidx.annotation.NonNull;
+import com.google.android.datatransport.Encoding;
+import java.util.Arrays;
+
+/**
+ * Represents encoded payloads.
+ *
+ * <p>It is essentially a pair of {@code (encoding, bytes)}, where {@code bytes} are encoded using
+ * {@code encoding}.
+ */
```
Comment (rlazo) [inline transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/EncodedPayload.java]: Also comment about the need for custom equals

File: transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/EncodedPayload.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  }
+
+  public byte[] getBytes() {
+    return bytes;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
```
Comment (rlazo) [inline transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/EncodedPayload.java]: `instanceof`

File: transport/transport-backend-cct/src/test/java/com/google/android/datatransport/cct/CctTransportBackendTest.java (hunk 1 above)
Comment (vkryachko) [reply transport/transport-backend-cct/src/test/java/com/google/android/datatransport/cct/CctTransportBackendTest.java]: Done

File: root-project.gradle [hunk 4]
```diff
@@ -100,7 +100,7 @@ configure(subprojects) {
     }
     task(['type': JacocoReport, 'dependsOn': 'check', 'group': 'Coverage',
           'description': 'Generates JaCoCo unit test coverage reports.'], checkCoverage) {
-        def excludes = ['**/R.class', '**/R$*.class', '**/BuildConfig.*', '**/proto/**', '**Manifest*.*']
+        def excludes = ['**/R.class', '**/R$*.class', '**/BuildConfig.*', '**/proto/**', '**Manifest*.*', '**/AutoValue_*.class', '**/*_*Factory.class']
```
Comment (rlazo) [inline root-project.gradle]: I'm not sure about these changes. When looking at the coverage reports it will seem like these `TransportRuntime` changes are related to the drop, when it's actually different. Consider moving them into their own PR.

File: transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CctTransportBackend.java [hunk 5]
```diff
@@ -197,6 +198,12 @@ private BatchedLogRequest getRequestBody(BackendRequest backendRequest) {
       }
 
       for (EventInternal eventInternal : entry.getValue()) {
+        Encoding encoding = eventInternal.getEncodedPayload().getEncoding();
+        if (!encoding.equals(Encoding.of("proto"))) {
+          Logging.d(LOG_TAG, "Received event of unsupported encoding %s. Skipping...", encoding);
```
Comment (rlazo) [inline transport/transport-backend-cct/src/main/java/com/google/android/datatransport/cct/CctTransportBackend.java]: Is it more `debug` than `warning`?
